src/main/java/org/jalasoft/moi/model/csharp/CsharpCommandBuilder.java

```diff
@@ -13,40 +13,27 @@
 import java.nio.file.Path;
 
 /**
- * Builds a command and execute a c# file using the path provided by Params object
- * and uses the complete path of c# in window to build its string
+ * Builds a command for a c# file compilation and execution
+ * using the path provided by Params object
  *
  * @author Carlos Meneses
  * @version 1.0
  */
 public class CsharpCommandBuilder implements ICommandBuilder {
 
     private static final String COMPILER_PATH = "C:/Windows/Microsoft.NET/Framework64/v4.0.30319/csc.exe ";
-    private static final String CHANGE_DIRECTORY = "cd ";
-
-    /**
-     * @param path contains the location of the directory or file
-     * @return String of the directory to change
-     */
-    private String getFolderPath(Path path) {
-        return CHANGE_DIRECTORY + path.getParent().toString();
-    }
+    private static final String COMPILE_ALL_AND_OUTPUT = "-optimize -out:Output.exe *.cs";
+    private static final String RUN_OUTPUT = "Output";
+    private static final String MOVE_TO = "cd ";
+    private static String folderPath;
 
     /**
-     * @param path contains the location of the directory or file
-     * @return String of the compiled file name
-     */
-    private String getCompiledName(Path path) {
-        return path.getFileName().toString().replace(".cs", ".exe");
-    }
-
-    /**
-     * @param path contains the location of the directory or file
-     * @return String of the command builded with the path Params
+     * @path completePath contains the location of the directory of the files
+     * @return String of the command builded with the path received
      */
     @Override
-    public String buildCommand(Path path) {
-        return getFolderPath(path)+" && "+COMPILER_PATH+path.getFileName().toString()+" && "+getCompiledName(path);
+    public String buildCommand(Path completePath) {
+        folderPath = completePath.toString();
+        return MOVE_TO + folderPath + " && " + COMPILER_PATH + COMPILE_ALL_AND_OUTPUT + " && " + RUN_OUTPUT;
```

Comment:
I believe that you should put this `completePath.toString()` directly into the command
`return MOVE_TO + completePath.toString() + " && " + CO...`